src/components/widgets/TimeRangeFilter/index.js

```diff
@@ -0,0 +1,156 @@
+}
+
+function inferSliderHandlebarLocations(injectTimeRange = false) {
+  return function dateToHandlebarLocation(props) {
+    const dynamicTimeRange = {
+      min: props.selectedTimeRange.start.diff(props.availableTimeRange.start,'months'),
+      max: props.selectedTimeRange.end.diff(props.availableTimeRange.start,'months'),
+    };
+
+    const injectedTimeRange = injectTimeRange
+      ? _.clone(dynamicTimeRange) : {min: null, max: null};
+
+    return {...props, dynamicTimeRange, injectedTimeRange};
+  };
+}
+
+function renderLabels(labels) {
+  //TODO, rename first label to "Now".
+  return (
+    <ul>
+      {labels.reduce((list, label) => {
+        if (list.length > 0) {
+          list.push(<li/>);
+        }
+        list.push(<li>{label}</li>);
+        return list;
+      },[])}
+    </ul>
+  );
+}
+
+function renderView(state$) {
+  return state$.map(state => {
+    const selectableMonthCount
+     = state.availableTimeRange.end.diff(state.availableTimeRange.start, 'months');
+
+    const labels = _.range(selectableMonthCount)
+      .map(m => state.availableTimeRange.start.clone()
+      .add(m,'month').format('MMM'));
+
+    // This should be same as in palette.scss
+    const colorGrayLighter = '#D8D8D8';
+
+    const p1 = state.dynamicTimeRange.min / (selectableMonthCount - 1);
+    const p2 = state.dynamicTimeRange.max / (selectableMonthCount - 1);
+
+    const sliderStyle = {
+      backgroundImage: `-webkit-gradient(
+        linear, left top, right top,
+        from(${colorGrayLighter}),
+        color-stop(${p1}, ${colorGrayLighter}),
+        color-stop(${p1}, black),
+        color-stop(${p2}, black),
+        color-stop(${p2}, ${colorGrayLighter}),
+        to(${colorGrayLighter})
+      )`,
+    };
+
+    return (
+      <div className={`TimeRangeFilter ${styles.timeRangeFilter}`}>
+        <p>Within this time frame</p>
+        <section>
+          <input
+            data-hook={new ControlledInputHook(state.injectedTimeRange.min)}
+            min="0"
+            max={selectableMonthCount - 1}
+            step="1"
+            type="range"
+          />
+          <input style={sliderStyle}
+            data-hook={new ControlledInputHook(state.injectedTimeRange.max)}
+            min="0"
+            max={selectableMonthCount - 1}
+            step="1"
+            type="range"
+          />
+         {renderLabels(labels)}
+        </section>
+      </div>
+    );
+  });
+}
+
+function TimeRangeFilter(sources) {
+  const actions = intent(sources.DOM);
+
+  const update$ = makeUpdate(actions, sources.props$);
+
+  // The delay here is a hack: otherwise both funcs will be called before rendering,
+  // causing hook to be null for the first render.
+  const state$ = Rx.Observable.merge(
+    sources.props$.first().map(inferSliderHandlebarLocations(true)),
+    model(sources.props$, update$).map(inferSliderHandlebarLocations()).delay(10)
+  );
+
```

Comment:
First, remove the `.delay(10)`. It looks rather hacky. The problem is that `model(sources.props$, update$).map(inferSliderHandlebarLocations())` emits initially because of those 2 startWith which are joined in combineLatest. You can fix this by putting `.skip(1)` after the combineLatest (inside makeUpdate function).